homeassistant/components/dublin_bus_transport/sensor.py

```diff
@@ -92,18 +80,11 @@ def state(self):
     def device_state_attributes(self):
         """Return the state attributes."""
         if self._times is not None:
-            next_up = "None"
-            if len(self._times) > 1:
-                next_up = f"{self._times[1][ATTR_ROUTE]} in "
-                next_up += self._times[1][ATTR_DUE_IN]
-
             return {
-                ATTR_DUE_IN: self._times[0][ATTR_DUE_IN],
-                ATTR_DUE_AT: self._times[0][ATTR_DUE_AT],
                 ATTR_STOP_ID: self._stop,
                 ATTR_ROUTE: self._times[0][ATTR_ROUTE],
+                ATTR_TIMETABLE: self._times,
```

Comment:
This should not be an attribute, if there is data here that is useful, the integration should set up multiple sensors instead (separate PR).

Reply:
Can you please elaborate a bit more here about multiple sensors ?

Reply:
`self._times` is a list of dictionaries, where this sensor now uses the first `0` element in the list.
Since you added `self._times` here I assume you wanted to show the rest of the items in the list as well?
If that information is desirable, the integration should then set up 1 sensor pr element in that list or maybe just 0 and 1?
Anyway, that should be a different PR if you decide to go down that route, for now, let's just keep it to the first element.

Reply:
That's correct `self._times` is a list of timetables. What I am trying to achieve here is to have a full timetable list of the desired bus/bus_stop besides the default sensor of the 'next bus' as not everybody (including me) find really useful. Having the full list of timetables as an extra attribute gives users more flexibility to plot as a 'real time' feeling of the bus_stop/bus route timetable. In other words, If I don't care about the next bus, I wanna know when is my bus+x time ahead as you do when you are at the bus station. 

> the integration should then set up 1 sensor pr element in that list or maybe just 0 and 1?
This is how it was before, but I believe having full timetable instead of the 'next item in the list' will be much appreciated by whoever will using it.

Reply:
Those should be separate entities.

Reply:
Can you give me more context here? why it should be a separate entities ? AFAIK there is no native way in lovelance to plot something like a table with various values. if you could point me to the an example/guidance documentation it would be awesome. Thank you

Reply:
Most other have the first departure as the state and the next departure as an attribute that would be fine here as well, not the entire list (which would give you "mini" sensors") inside the attributes of a sensor.

The backend should not care how the frontend works.

Reply:
> Most other have the first departure as the state and the next departure as an attribute that would be fine here as well,

yep, that's what surprised me though :)  Why not having the full timetable instead of only the next departure or 2? This is pointless if you need to know when is the next transport in 30 minutes if they run at random times.  I mean whoever needs only next departure fine, but what about ppl who requires more than that ? In this case all of the data is gathered within 1 api call and the attribute data is really nicely plotted with [flex-table-card](https://github.com/custom-cards/flex-table-card) 
Not pressing or pushing just trying to come with arguments why i'm in favor of timetable instead of next departure entry only. I believe the full timetable is nice to have and it should be available out of the box so people can use it and not write custom stuff. Why having next departure as an attribute is ok, but a list of the next departures is not ok ? 

>The backend should not care how the frontend works.

Correct, it doesn't, but the backend expose the data what can be rendered in frontend :)

I just want to share with the community what I find useful and make this available out of the box :)